### ISSUE TITLE
`OneWayANOVATest` hang forever for all zero data

### ISSUE THREAD

**Issue by babaq** [JuliaStats/HypothesisTests.jl#301]
```julia
using HypothesisTests
a=[fill(0.0,20) for _ in 1:5]
h=OneWayANOVATest(a...)
```
REPL shows following and hang here.
Population details:
    parameter of interest:   Means
    value under h_0:         "all equal"
    point estimate:          NaN

I am using Julia v1.9.1 with HypothesisTests v0.11 on x64 Win10.

**Comment by ararslan** [comment]
Hm, I can't reproduce with Julia v1.9.1 on macOS ARM. I suspect this is related to a bug that was fixed in the most recent release of SpecialFunctions. What version does `]status -m SpecialFunctions` show for you? If it's anything other than v2.3.0, try doing `]update`.

**Comment by babaq** [comment]
you are right, problem solved after updating SpecialFunctions to v2.3. 
Thanks.
